From 6febf9854c41c710177ac1b0f29d0ed585ab2738 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 12 May 2021 10:37:39 +0300
Subject: [PATCH] Revert "[SW-41316] preboot: make flexible fw load setting platform specific"

Since dynamic load capability is fixed for greco, this patch
reverts changes to set CPU_BOOT_DEV_STS0_FW_LD_COM_EN bit platform
specific and enable for all platforms.

This reverts commit f1df7bb18977ed65b98947efe3b7ea9ac3a2ecd7.

Change-Id: I83ffd882f8673b9991bc6f674ec8560377c23a78
---

diff --git a/gaudi/common.c b/gaudi/common.c
index f6624d3..3c883d1 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -418,7 +418,6 @@
 void lkd_dev_specific_setup(void)
 {
 	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN, true);
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
 }
 
 /* Hard coded each of the gaudi on-chip TS device properties */
diff --git a/goya/common.c b/goya/common.c
index 684fef7..6a28545 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -25,7 +25,6 @@
 #include <preboot/fit.h>
 #include <preboot/thermal.h>
 #include <string.h>
-#include <boot_if.h>
 
 #include "common.h"
 
@@ -221,11 +220,6 @@
 u32 hl_pll_timestamp_base_get(void)
 {
 	return mmCPU_TIMESTAMP_CNTCR;
-}
-
-void lkd_dev_specific_setup(void)
-{
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
 }
 
 #define GOYA_ASIC_TS_MAX_COUNT 3
diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index b6c84bd..ab895c8 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -32,6 +32,7 @@
 
 void module_lkd_enable(void)
 {
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
 	cpu_boot_sts_set(CPU_BOOT_STATUS_WAITING_FOR_BOOT_FIT);
 }
 
